ext/network.go

```diff
@@ -0,0 +1,488 @@
+//
+// IP objects support various inspection methods.
+//
+//	<ip>.family() -> int
+//	<ip>.isLoopback() -> bool
+//	<ip>.isGlobalUnicast() -> bool
+//	<ip>.isLinkLocalMulticast() -> bool
+//	<ip>.isLinkLocalUnicast() -> bool
+//	<ip>.isUnspecified() -> bool
+//
+// The `ip.isCanonical` function takes a string and returns true if it matches
+// the RFC 5952 canonical string representation of that address.
+//
+//	ip.isCanonical(string) -> bool
+//
+// Examples:
+//
+//	ip('127.0.0.1').family() == 4
+//	ip('::1').family() == 6
+//	ip('127.0.0.1').isLoopback() == true
+//	ip.isCanonical('2001:db8::1') == true  // RFC 5952 format
+//	ip.isCanonical('2001:DB8::1') == false // Uppercase is not canonical
+//	ip.isCanonical('2001:db8:0:0:0:0:0:1') == false // Expanded is not canonical
+//
+// # CIDR Member Functions
+//
+// CIDR objects support containment checks and property extraction.
+//
+//	<cidr>.containsIP(ip|string) -> bool
+//	<cidr>.containsCIDR(cidr|string) -> bool
+//	<cidr>.ip() -> ip
+//	<cidr>.masked() -> cidr
+//	<cidr>.prefixLength() -> int
+//
+// Examples:
+//
+//	cidr('10.0.0.0/8').containsIP(ip('10.0.0.1')) == true
+//	cidr('10.0.0.0/8').containsIP('10.0.0.1') == true
+//	cidr('10.0.0.0/8').containsCIDR('10.1.0.0/16') == true
+//	cidr('192.168.1.5/24').ip() == ip('192.168.1.5')
+//	cidr('192.168.1.5/24').masked() == cidr('192.168.1.0/24')
+//	cidr('192.168.1.0/24').prefixLength() == 24
+func Network() cel.EnvOption {
+	return func(e *cel.Env) (*cel.Env, error) {
+		// Install the library (Types and Functions)
+		e, err := cel.Lib(&networkLib{})(e)
+		if err != nil {
+			return nil, err
+		}
+
+		// Install the Adapter (Wrapping the existing one)
+		adapter := &networkAdapter{Adapter: e.CELTypeAdapter()}
+		return cel.CustomTypeAdapter(adapter)(e)
+	}
+}
+
+const (
+	// Function names
+	isIPFunc             = "isIP"
+	isCIDRFunc           = "isCIDR"
+	ipFunc               = "ip"
+	cidrFunc             = "cidr"
+	familyFunc           = "family"
+	isCanonicalFunc      = "ip.isCanonical"
+	isLoopbackFunc       = "isLoopback"
+	isGlobalUnicastFunc  = "isGlobalUnicast"
+	isUnspecifiedFunc    = "isUnspecified"
+	isLinkLocalMcastFunc = "isLinkLocalMulticast"
+	isLinkLocalUcastFunc = "isLinkLocalUnicast"
+	containsIPFunc       = "containsIP"
+	containsCIDRFunc     = "containsCIDR"
+	maskedFunc           = "masked"
+	prefixLengthFunc     = "prefixLength"
+	ipFromCIDRFunc       = "ip"
+)
+
+var (
+	// Definitions for the Opaque Types
+	networkIPType   = types.NewOpaqueType("network.IP")
+	networkCIDRType = types.NewOpaqueType("network.CIDR")
+)
+
+type networkLib struct{}
+
+func (*networkLib) LibraryName() string {
+	return "cel.lib.ext.network"
+}
+
+func (*networkLib) CompileOptions() []cel.EnvOption {
+	return []cel.EnvOption{
+		// 1. Register Types
+		cel.Types(
+			networkIPType,
+			networkCIDRType,
+		),
+
+		// 2. Register Functions
+		cel.Function(isIPFunc,
+			cel.Overload("isIP_string", []*cel.Type{cel.StringType}, cel.BoolType,
```

Comment:
For the greatest degree of compatibility at the AST level, you'll need to make sure that the function and overload names from K8s are copied here exactly, e.g. swap `ip_isLoopback` with `ip_is_loopback` ... there are a few others where the casing doesn't match.